Traceback (most recent call last):
  File "/usr/share/miniconda3/envs/runenv/lib/python3.10/site-packages/jupyter_cache/executors/utils.py", line 51, in single_nb_execution
    executenb(
  File "/usr/share/miniconda3/envs/runenv/lib/python3.10/site-packages/nbclient/client.py", line 1204, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/usr/share/miniconda3/envs/runenv/lib/python3.10/site-packages/nbclient/util.py", line 84, in wrapped
    return just_run(coro(*args, **kwargs))
  File "/usr/share/miniconda3/envs/runenv/lib/python3.10/site-packages/nbclient/util.py", line 62, in just_run
    return loop.run_until_complete(coro)
  File "/usr/share/miniconda3/envs/runenv/lib/python3.10/asyncio/base_events.py", line 649, in run_until_complete
    return future.result()
  File "/usr/share/miniconda3/envs/runenv/lib/python3.10/site-packages/nbclient/client.py", line 663, in async_execute
    await self.async_execute_cell(
  File "/usr/share/miniconda3/envs/runenv/lib/python3.10/site-packages/nbclient/client.py", line 965, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/usr/share/miniconda3/envs/runenv/lib/python3.10/site-packages/nbclient/client.py", line 862, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
img = plt.imread("https://hackage.haskell.org/package/JuicyPixels-extra-0.1.0/src/data-examples/lenna.png")
plt.imshow(img)
------------------

[0;31m---------------------------------------------------------------------------[0m
[0;31mValueError[0m                                Traceback (most recent call last)
Cell [0;32mIn[20], line 1[0m
[0;32m----> 1[0m img [38;5;241m=[39m [43mplt[49m[38;5;241;43m.[39;49m[43mimread[49m[43m([49m[38;5;124;43m"[39;49m[38;5;124;43mhttps://hackage.haskell.org/package/JuicyPixels-extra-0.1.0/src/data-examples/lenna.png[39;49m[38;5;124;43m"[39;49m[43m)[49m
[1;32m      2[0m plt[38;5;241m.[39mimshow(img)

File [0;32m/usr/share/miniconda3/envs/runenv/lib/python3.10/site-packages/matplotlib/pyplot.py:2123[0m, in [0;36mimread[0;34m(fname, format)[0m
[1;32m   2121[0m [38;5;129m@_copy_docstring_and_deprecators[39m(matplotlib[38;5;241m.[39mimage[38;5;241m.[39mimread)
[1;32m   2122[0m [38;5;28;01mdef[39;00m [38;5;21mimread[39m(fname, [38;5;28mformat[39m[38;5;241m=[39m[38;5;28;01mNone[39;00m):
[0;32m-> 2123[0m     [38;5;28;01mreturn[39;00m [43mmatplotlib[49m[38;5;241;43m.[39;49m[43mimage[49m[38;5;241;43m.[39;49m[43mimread[49m[43m([49m[43mfname[49m[43m,[49m[43m [49m[38;5;28;43mformat[39;49m[43m)[49m

File [0;32m/usr/share/miniconda3/envs/runenv/lib/python3.10/site-packages/matplotlib/image.py:1536[0m, in [0;36mimread[0;34m(fname, format)[0m
[1;32m   1532[0m img_open [38;5;241m=[39m (
[1;32m   1533[0m     PIL[38;5;241m.[39mPngImagePlugin[38;5;241m.[39mPngImageFile [38;5;28;01mif[39;00m ext [38;5;241m==[39m [38;5;124m'[39m[38;5;124mpng[39m[38;5;124m'[39m [38;5;28;01melse[39;00m PIL[38;5;241m.[39mImage[38;5;241m.[39mopen)
[1;32m   1534[0m [38;5;28;01mif[39;00m [38;5;28misinstance[39m(fname, [38;5;28mstr[39m) [38;5;129;01mand[39;00m [38;5;28mlen[39m(parse[38;5;241m.[39murlparse(fname)[38;5;241m.[39mscheme) [38;5;241m>[39m [38;5;241m1[39m:
[1;32m   1535[0m     [38;5;66;03m# Pillow doesn't handle URLs directly.[39;00m
[0;32m-> 1536[0m     [38;5;28;01mraise[39;00m [38;5;167;01mValueError[39;00m(
[1;32m   1537[0m         [38;5;124m"[39m[38;5;124mPlease open the URL for reading and pass the [39m[38;5;124m"[39m
[1;32m   1538[0m         [38;5;124m"[39m[38;5;124mresult to Pillow, e.g. with [39m[38;5;124m"[39m
[1;32m   1539[0m         [38;5;124m"[39m[38;5;124m``np.array(PIL.Image.open(urllib.request.urlopen(url)))``.[39m[38;5;124m"[39m
[1;32m   1540[0m         )
[1;32m   1541[0m [38;5;28;01mwith[39;00m img_open(fname) [38;5;28;01mas[39;00m image:
[1;32m   1542[0m     [38;5;28;01mreturn[39;00m (_pil_png_to_float_array(image)
[1;32m   1543[0m             [38;5;28;01mif[39;00m [38;5;28misinstance[39m(image, PIL[38;5;241m.[39mPngImagePlugin[38;5;241m.[39mPngImageFile) [38;5;28;01melse[39;00m
[1;32m   1544[0m             pil_to_array(image))

[0;31mValueError[0m: Please open the URL for reading and pass the result to Pillow, e.g. with ``np.array(PIL.Image.open(urllib.request.urlopen(url)))``.
ValueError: Please open the URL for reading and pass the result to Pillow, e.g. with ``np.array(PIL.Image.open(urllib.request.urlopen(url)))``.

